tools/swift-inspect/Sources/SwiftInspectLinux/ElfFile.swift

```diff
@@ -0,0 +1,144 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2024 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import Foundation
+import LinuxSystemHeaders
+
+// TODO: replace this implementation with general purpose ELF parsing support
+// currently private to swift/stdlib/public/Backtrace.
+class ElfFile {
+  public enum ELFError: Error {
+    case notELF64(_ filePath: String, _ description: String = "")
+    case malformedFile(_ filePath: String, _ description: String = "")
+  }
+
+  public typealias SymbolMap = [String: (start: UInt64, end: UInt64)]
+
+  let filePath: String
+  let fileData: Data
+  let ehdr: Elf64_Ehdr
+
+  public init(filePath: String) throws {
+    self.filePath = filePath
+
+    let fileData = try Data(contentsOf: URL(fileURLWithPath: filePath), options: .alwaysMapped)
+    self.fileData = fileData
+
+    let identLen = Int(EI_NIDENT)
+    let identMagic = String(bytes: fileData[0..<identLen].prefix(Int(SELFMAG)), encoding: .utf8)
+    guard identMagic == ELFMAG else {
+      throw ELFError.notELF64(filePath)
+    }
+
+    let identClass = fileData[Int(EI_CLASS)]
```

Comment:
Should this not operate on the identity buffer rather than indexing into the file data? It _just_ happens that the base indicies are equal and this works.